### ISSUE TITLE
Lamar 2.0.2 error on startup

### ISSUE THREAD

**Issue by joergjo** [JasperFx/lamar#119]
`Lamar.Microsoft.DependencyInjection 2.0.2` fails on startup for a simple ASP.NET Core MVC 2.2 application.

### Create project
```
mkdir LamarRepro
cd LamarRepro
dotnet new mvc
dotnet add package Lamar.Microsoft.DependencyInjection -v 2.0.2
```
### Enable Lamar as IoC container:
- Edit `Program.cs` to call `UseLamar()` before `UseStartup<Startup>()`
- Replace `public void ConfigureServices(IServiceCollection services)` in `Startup.cs` with `public void ConfigureContainer(ServiceRegistry services)` 

### Run it
```
dotnet build
dotnet run
```
The app immediately crashes with
```
Unhandled Exception: System.MissingMethodException: Method not found: 'Void LamarCompiler.Frames.ConstructorFrame..ctor(System.Type, System.Reflection.ConstructorInfo, System.Func`2<LamarCompiler.Frames.ConstructorFrame,LamarCompiler.Model.Variable>)'.
   at Lamar.IoC.Instances.InstanceConstructorFrame..ctor(ConstructorInstance instance, DisposeTracking disposal, Variable[] arguments, SetterArg[] setterParameters)
   at Lamar.IoC.Instances.ConstructorInstance.CreateBuildFrame()
   at Lamar.IoC.Instances.GeneratedInstance.GenerateResolver(GeneratedAssembly generatedAssembly)
   at Lamar.IoC.Instances.GeneratedInstance.buildResolver(Scope scope)
   at Lamar.IoC.Instances.GeneratedInstance.ToResolver(Scope topScope)
   at Lamar.ServiceGraph.FindResolver(Type serviceType)
   at Lamar.IoC.Scope.TryGetInstance(Type serviceType)
   at Microsoft.Extensions.DependencyInjection.ActivatorUtilities.GetServiceOrCreateInstance(IServiceProvider provider, Type type)
   at Microsoft.AspNetCore.Hosting.Internal.StartupLoader.LoadMethods(IServiceProvider hostingServiceProvider, Type startupType, String environmentName)
   at Microsoft.AspNetCore.Hosting.WebHostBuilderExtensions.<>c__DisplayClass1_0.<UseStartup>b__1(IServiceProvider sp)
   at Lamar.IoC.Resolvers.SingletonResolver`1.Resolve(Scope scope)
   at Lamar.ServiceGraph.<>c__DisplayClass46_1.<FindResolver>b__0(Scope s)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetService[T](IServiceProvider provider)
   at Microsoft.AspNetCore.Hosting.Internal.WebHost.EnsureStartup()
   at Microsoft.AspNetCore.Hosting.Internal.WebHost.EnsureApplicationServices()
   at Microsoft.AspNetCore.Hosting.Internal.WebHost.Initialize()
   at Microsoft.AspNetCore.Hosting.WebHostBuilder.Build()
   at LamarRepro.Program.Main(String[] args) in C:\Users\joergjo\source\Repos\LamarRepro\Program.cs:line 18
```

The same works when downgrading to v2.0.1.

**Comment by Megasware128** [comment]
The wrong LamarCompiler is being depend on.
![image](https://user-images.githubusercontent.com/3170510/51168498-c4b13c00-18a9-11e9-939e-1b0533d3d922.png)
Workaround is `dotnet add package LamarCompiler -v 1.0.1`

**Comment by joergjo** [comment]
Good catch, will try that!

**Comment by joergjo** [comment]
Fixed in Lamar 2.0.4.

**Comment by jeremydmiller** [comment]
Closing....
